src/WebJobs.Script/Rpc/LanguageWorkerChannelInitializationService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class LanguageWorkerChannelInitializationService : IHostedService
```

Comment:
With these being initialize at process level rather than host level, they'll be initialized even when the host is in Offline mode. The intent of Offline mode was to ensure that nothing from the bin directory is loaded (e.g. to allow extensions to be installed). Where are custom language workers installed? They're part of our site extension package and not in bin dir?

Reply:
Good point. Workers are part of the bin dir. Should I check if app is in offline mode before adding?

Reply:
When in Offline mode, anything that might be holding locks on the user bin dir must be unloaded, putting the host in a state where it can install extensions and modify files w/o locking issues.

Reply:
Thanks!